src/main/java/teammates/storage/api/ProfilesDb.java

```diff
@@ -186,7 +186,7 @@ private StudentProfile getCurrentProfileFromDb(String googleId)
 
     private void ensureUpdatingProfileExists(String googleId,
             StudentProfile profileToUpdate) throws EntityDoesNotExistException {
-        if (profileToUpdate == null) {
+        if (profileToUpdate == null || JDOHelper.isDeleted(profileToUpdate)) {
```

Comment:
Hmm actually don't need to check here too coz the profileToUpdated that is passed in is retrieved from getStudentProfileEntityFromDb which already checked. Removing the check here.